src/backend/controllers/battleController.js

```diff
@@ -0,0 +1,249 @@
+        return res
+            .status(404)
+            .json({ msg: "Could not find the personBeingChallenged's tank in DB"});
+    }
+
+    const challengerTank = await Tank.findById(challengerTankId);
+
+    if(challengerTank == null){
+        return res
+            .status(404)
+            .json({ msg: "Could not find the challenger's Tank in DB"});
+    }
+
+    
+    // Query the amount of money the challenger has
+    const challengerUserDoc = await User.findById(challengerTank.userId, 'money');
+    
+    if(challengerUserDoc == null){
+        return res
+            .status(404)
+            .json({ msg: 'Could not find the challenger in DB'});
+    }
+
+    if(challengerUserDoc.money < personBeingChallengedUserDoc.wager){
+        return res
+            .status(401)
+            .json({ msg: 'Challenger does not have enough money to wager'});
+    }
+
+    const newRecord = new BattleRecord({
+        userOne: personBeingChallengedId,
+        userTwo: challengerTank.userId,
+        tankOne: personBeingChallengedTank,
+        tankTwo: challengerTank,
+        winner: -1,
+        prizeMoney: (personBeingChallengedUserDoc.wager * 2), // Each person puts in for the wager
+        eloExchanged: 0
+    });
+    
+    // Take the wager amount from the challenger's money
+    const challengerBalance = challengerUserDoc.money - personBeingChallengedUserDoc.wager;
+    challengerUserDoc.money = challengerBalance;
+
+    // Save the updated balance to the db
+    await challengerUserDoc.save();
+    
+    await newRecord.save ((err: Error) => {
+        if (err) {
+            console.error(err.message);
+            return res
+                .status(500)
+                .json({ msg: 'Unable to save battle record to DB.' });
+        }
+        else{
+            console.log('Match successfully created!');
+            return res
+                .status(200)
+                .send(newRecord._id)
+        }
+    });
+   } catch (err) {
+       return res
+               .status(500)
+               .json({ msg: 'Server Error'});
+   }
+}
+
+exports.reportResults = async (req: Request, res: Response) => 
+{
+    // Checks that we have received the correct body from the frontend 
+    const errors = validationResult(req);
+
+   if (!errors.isEmpty()) {
+       // Return 400 for a bad request
+       return res
+           .status(400)
+           .json({ errors: errors.array() });
+   }
+
+   const { winner, battleId } = req.body;
+
+   try {
+    const battle = await BattleRecord.findByIdAndUpdate(battleId, { winner : winner }, {new: true}, (err: Error) => {
+        if (err) {
+            console.error(err.message);
+            return res
+             .status(500)
+             .json({ msg: 'Failed up update battleRecord'});
+        }
+    });
+ 
+    if (battle == null) {
+        console.log('battle not found');
+        return res
+             .status(404)
+             .json({ msg: 'Could not find battle'});
+    }
+ 
```

Comment:
We only want to do all of these updates if the previous battle had no defined winner. If Alice creates the match and then both Alice and Bob watch it before the other finishes, then the money will get added back to both player's accounts twice. If the previous battle record already has a reported winner, we can just return a success message early if the winners match, or print something on the server if the winners don't match.

Reply:
Check if a winner has already been reported and if it has check if the winners match and return. If they dont match report an error.